### ISSUE TITLE
mnn 跑bert 出现CL ERROR CODE : -58, info:clEvent  CL ERROR CODE : -14, info:raster_buffer

### ISSUE THREAD

**Issue by dshwei** [alibaba/MNN#2315]
## 平台
Processor	12th Gen Intel(R) Core(TM) i7-12700   2.10 GHz
Installed RAM	16.0 GB

## mnn
2.4.0

# opencl出错

The device support dot:0, support fp16:0, support i8mm: 0
MNN use high precision
Map error biasPtrCL == nullptr 
Map error biasPtrCL == nullptr 
Map error biasPtrCL == nullptr 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer 
lws tune res raster_buffer
CL ERROR CODE : -58, info:clEvent 
CL ERROR CODE : -14, info:raster_buffer

**Comment by bitxsw93** [comment]
1、问下使用的是哪个版本的MNN代码？
2、附上机型、模型，以及复现步骤？
3、Map error biasPtrCL == nullptr
查看一下这个报错，enqueueMapBuffer对应的返回值是多少？

**Comment by dshwei** [comment]
> 


MNN_GPU_TUNING_FAST   = 1 << 4
MNN_GPU_MEMORY_BUFFER = 1 << 6
MNN_GPU_MEMORY_IMAGE  = 1 << 7
index_to_label_ids =  {0:"news_story",  1:"news_culture", 2:"news_entertainment", 3:"news_sports",  4:"news_finance",  5:"news_house",  6:"news_car",  
7:"news_edu",  8:"news_tech",  9:"news_military",  10:"news_travel",  11:"news_world",  12:"stock",  13:"news_agriculture",  14:"news_game"}

def mode_run(model_path,sentence,tokenizer,data_dir=None):

    np.random.seed(0)
    interpreter = MNN.Interpreter(model_path)
    
    interpreter.setSessionMode(8)
    interpreter.setSessionHint(0, 20)

    config = {}
    config['backend']   = "OPENCL"
    config['precision'] = "high"
    config['numThread'] = MNN_GPU_TUNING_FAST | MNN_GPU_MEMORY_BUFFER # Buffer Mode
    session = interpreter.createSession(config)

    inputs = interpreter.getSessionInputAll(session)
    # 输入 resizeTensor
    interpreter.resizeTensor(inputs["attention_mask"], (1,128))
    interpreter.resizeTensor(inputs["token_type_ids"], (1,128))
    interpreter.resizeTensor(inputs["input_ids"], (1,128))
    # 重新计算形状分配内存
    interpreter.resizeSession(session)

    input_ids,attention_mask,token_type_ids = tokenizer.single_sentence_convert_to_feature(sentence)
    inference(interpreter,session,inputs,input_ids,attention_mask,token_type_ids)
    if data_dir:
        features  = tokenizer.dataset_convert_to_features(data_dir)
        for input_ids,attention_mask,token_type_ids in features:
           inference(interpreter,session,inputs,input_ids,attention_mask,token_type_ids)

def inference(interpreter,session,inputs,input_ids,attention_mask,token_type_ids):
    
    input_ids_tensor = MNN.Tensor(inputs["input_ids"].getShape(), MNN.Halide_Type_Int,np.array(input_ids).astype(np.int32), MNN.Tensor_DimensionType_Caffe)
    inputs["input_ids"].copyFrom(input_ids_tensor)

    attention_mask_tensor = MNN.Tensor(inputs["attention_mask"].getShape(), MNN.Halide_Type_Int,np.array(attention_mask).astype(np.int32), MNN.Tensor_DimensionType_Caffe)
    inputs["attention_mask"].copyFrom(attention_mask_tensor)


    token_type_ids_tensor = MNN.Tensor(inputs["token_type_ids"].getShape(), MNN.Halide_Type_Int,np.array(token_type_ids).astype(np.int32), MNN.Tensor_DimensionType_Caffe)
    inputs["token_type_ids"].copyFrom(token_type_ids_tensor)

    tmp_tensor = np.zeros((1,15),dtype=np.float32)
    interpreter.runSession(session)

    outputs = interpreter.getSessionOutput(session)

    out_tensor = MNN.Tensor(outputs.getShape(), MNN.Halide_Type_Float, tmp_tensor, MNN.Tensor_DimensionType_Caffe)

    outputs.copyToHostTensor(out_tensor)

    data = out_tensor.getData()
    index  = np.argmax(data)
    print("prediction result is: ",get_label_descrb(index))

**Comment by dshwei** [comment]
> 

mnn模型怎么发给你

**Comment by bitxsw93** [comment]
1、问下使用的是哪个版本的MNN代码？
2、是什么机型，使用MNNV2Basic.out试下可以复现吗？
3、Map error biasPtrCL == nullptr
查看一下这个报错，enqueueMapBuffer对应的返回值是多少？

**Comment by dshwei** [comment]
> 
已经添加到问题中了，
mnn模型怎么发给你

**Comment by jxt1234** [comment]
相关问题已经修正，更新版本后测试后仍有问题重新提 issue
